lib/Sema/CSGen.cpp

```diff
@@ -3874,35 +3874,38 @@ static Expr *generateConstraintsFor(ConstraintSystem &cs, Expr *expr,
 /// \param wrappedVar The property that has a property wrapper.
 /// \returns the type of the property.
 static Type generateWrappedPropertyTypeConstraints(
-   ConstraintSystem &cs, Type initializerType,
-   VarDecl *wrappedVar, ConstraintLocator *locator) {
+   ConstraintSystem &cs, Type initializerType, VarDecl *wrappedVar) {
   auto dc = wrappedVar->getInnermostDeclContext();
 
   Type wrapperType = LValueType::get(initializerType);
   Type wrappedValueType;
 
-  for (unsigned i : indices(wrappedVar->getAttachedPropertyWrappers())) {
+  auto wrapperAttributes = wrappedVar->getAttachedPropertyWrappers();
+  for (unsigned i : indices(wrapperAttributes)) {
     Type rawWrapperType = wrappedVar->getAttachedPropertyWrapperType(i);
-    if (!rawWrapperType || rawWrapperType->hasError())
+    auto wrapperInfo = wrappedVar->getAttachedPropertyWrapperTypeInfo(i);
+    if (rawWrapperType->hasError() || !wrapperInfo)
       return Type();
 
     // The former wrappedValue type must be equal to the current wrapper type
     if (wrappedValueType) {
+      auto *typeRepr = wrapperAttributes[i]->getTypeRepr();
+      auto *locator =
+          cs.getConstraintLocator(typeRepr, LocatorPathElt::ContextualType());
       wrapperType = cs.openUnboundGenericTypes(rawWrapperType, locator);
-      cs.addConstraint(ConstraintKind::Equal, wrappedValueType, wrapperType,
+      cs.addConstraint(ConstraintKind::Equal, wrapperType, wrappedValueType,
```

Comment:
`Equal` constraint supposed to be commutative, is this not the case or this ordering just makes more sense visually?

Reply:
You're right - this ordering is purely for the order the types appear in the contextual failure error message

Reply:
Sounds good!